Core.DomainServices/SystemUsage/ItSystemUsageOverviewReadModelUpdate.cs

```diff
@@ -129,10 +129,9 @@ private void PatchSensitiveDataLevels(ItSystemUsage source, ItSystemUsageOvervie
         private static void PatchMainContract(ItSystemUsage source, ItSystemUsageOverviewReadModel destination)
         {
             destination.MainContractId = source.MainContract?.ItContractId;
-            destination.MainContractName = source.MainContract?.ItContract.Name;
-
-            destination.MainContractSupplierId = source.MainContract?.ItContract.Supplier?.Id;
-            destination.MainContractSupplierName = source.MainContract?.ItContract.Supplier?.Name;
+            destination.MainContractSupplierId = source.MainContract?.ItContract?.Supplier?.Id;
+            destination.MainContractSupplierName = source.MainContract?.ItContract?.Supplier?.Name;
```

Comment:
Hvis der er en MainContract så er det da krævet at der er en kontrakt på? Er det for læsbarhedens skyld eller er der en anden grund til at der er ? på alle delene?

Reply:
tja det er jo en antagelse der kan gå i stykker og beror på dyb viden om db. Det skader ikke med det ekstra sikkerhedsnet. Du kan grundlæggende ikke se ud af koden at den er sat med garanti. Det ved du kun pga. skemaet :)